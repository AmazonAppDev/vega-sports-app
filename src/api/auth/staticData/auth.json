{
  "auth": {
    "id": "nmbsd-234sdc-678-jklsdf",
    "name": "Freida",
    "surname": "Gomam",
    "email": "fgomam@fakemail.com",
    "address": "Nowhere 5 USA",
    "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/5p.jpg",
    "profiles": [
      {
        "id": "bada55-1dl00k-4tm33",
        "displayName": "Freida",
        "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/6p.jpg"
      },
      {
        "id": "l00k-1nt0-s0uL",
        "displayName": "Jennifer",
        "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/7p.jpg"
      }
    ]
  }
}
