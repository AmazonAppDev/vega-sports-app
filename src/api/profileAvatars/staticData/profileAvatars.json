{
  "profileavatars": [
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/1p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/2p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/3p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/4p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/5p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/6p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/7p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/8p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/9p.jpg"
  ]
}
