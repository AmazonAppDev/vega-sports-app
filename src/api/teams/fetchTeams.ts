import { useQuery } from '@tanstack/react-query';

import {
  ApiClient,
  isSuccessResponse,
  Endpoints,
} from '@AppServices/apiClient';
import type { TeamsDto } from './dtos/TeamsDto';
import { parseTeamsDtoArray } from './dtos/TeamsDto';
import staticData from './staticData/teams.json';

type ResponseDto = TeamsDto[];

const endpoint = Endpoints.Teams;

export const fetchTeamsApiCall = async () => {
  const response = await ApiClient.get<ResponseDto>(
    endpoint,
    { staticData },
    { isAuthorized: true },
  );

  if (!isSuccessResponse(response)) {
    switch (response.status) {
      case 400:
        throw new Error(
          `fetchTeamsApiCall(): resources does not exists for endpoint '${endpoint}'`,
        );

      default:
        throw new Error(
          `fetchTeamsApiCall(): failed to fetch data from endpoint '${endpoint}'`,
        );
    }
  }

  return parseTeamsDtoArray(response.data);
};

export const useTeams = () => {
  const query = useQuery({
    queryKey: [endpoint],
    queryFn: fetchTeamsApiCall,
  });

  return query;
};
