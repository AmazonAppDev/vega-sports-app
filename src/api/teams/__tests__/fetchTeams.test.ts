import { useQuery } from '@tanstack/react-query';
import { renderHook } from '@testing-library/react-native';

import { AppConfig } from '@AppServices/appConfig';
import { useTeams, fetchTeamsApiCall } from '../fetchTeams';
import staticData from '../staticData/teams.json';

jest.mock('@tanstack/react-query');

const mockSWR = useQuery as jest.Mock;

describe('fetchTeams', () => {
  beforeAll(() => {
    require('jest-fetch-mock').enableMocks();
  });

  afterEach(() => {
    fetchMock.resetMocks();
  });

  describe('fetchTeamsApiCall', () => {
    it('should return parsed team data on success', async () => {
      fetchMock.mockResponseOnce(JSON.stringify({ data: staticData.teams }), {
        status: 200,
      });

      const result = await fetchTeamsApiCall();

      expect(result).toMatchSnapshot();
    });

    it('should throw error for 400 status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      fetchMock.mockResponseOnce(JSON.stringify({}), { status: 400 });

      await expect(fetchTeamsApiCall()).rejects.toThrow(
        `fetchTeamsApiCall(): resources does not exists for endpoint 'teams'`,
      );
    });

    it('should throw error for unexpected Api status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      // @ts-expect-error intentionally break TS
      fetchMock.mockResponseOnce(null);

      await expect(fetchTeamsApiCall()).rejects.toThrow(
        `Perform a request failed`,
      );
    });

    it('should throw error for other non-200 status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      fetchMock.mockResponseOnce(JSON.stringify({}), { status: 500 });

      await expect(fetchTeamsApiCall()).rejects.toThrow(
        `fetchTeamsApiCall(): failed to fetch data from endpoint 'teams'`,
      );
    });
  });

  describe('useTeams', () => {
    it('should return all teams data and no error on successful fetch', async () => {
      fetchMock.mockResponseOnce(JSON.stringify({ data: staticData.teams }), {
        status: 200,
      });

      mockSWR.mockReturnValueOnce({
        data: await fetchTeamsApiCall(),
        error: null,
        isLoading: false,
      });

      const { result } = renderHook(() => useTeams());

      expect(result.current.data).toMatchSnapshot();
      expect(result.current.error).toBe(null);
      expect(result.current.isLoading).toBe(false);
    });

    it('should return isLoading as true before data loads', () => {
      mockSWR.mockReturnValueOnce({
        data: null,
        error: null,
        isLoading: true,
      });

      const { result } = renderHook(() => useTeams());

      expect(result.current.isLoading).toBe(true);
      expect(result.current.data).toBe(null);
      expect(result.current.error).toBe(null);
    });

    it('should return error when fetch fails', () => {
      const mockError = new Error('Network error');

      mockSWR.mockReturnValueOnce({
        data: null,
        error: mockError,
        isLoading: false,
      });

      const { result } = renderHook(() => useTeams());

      expect(result.current.error).toBe(mockError);
      expect(result.current.data).toBe(null);
      expect(result.current.isLoading).toBe(false);
    });
  });
});
