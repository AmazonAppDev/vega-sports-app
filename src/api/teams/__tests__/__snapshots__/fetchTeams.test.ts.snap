// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchTeams fetchTeamsApiCall should return parsed team data on success 1`] = `
[
  {
    "favorite": true,
    "itemId": "be00b94a-e12a-48c4-9615-23c3478f1bd0",
    "teamLogo": "https://robohash.org/iureinipsam.png?size=50x50&set=set1",
    "teamName": "Silver Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/6.jpg",
    "title": "Silver Team",
  },
  {
    "favorite": true,
    "itemId": "fdd82ada-af3f-4fe8-a90b-254877082e97",
    "teamLogo": "https://robohash.org/odiotemporeaccusamus.png?size=50x50&set=set1",
    "teamName": "Sunset Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/8.jpg",
    "title": "Sunset Team",
  },
  {
    "favorite": false,
    "itemId": "635e3d02-f128-460d-a630-628fec86ef2e",
    "teamLogo": "https://robohash.org/omnisadsequi.png?size=50x50&set=set1",
    "teamName": "Liberty Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/10.jpg",
    "title": "Liberty Team",
  },
  {
    "favorite": true,
    "itemId": "f6dd98e2-d7d2-48bb-ab4d-251a0c21255a",
    "teamLogo": "https://robohash.org/molestiaevoluptasdeserunt.png?size=50x50&set=set1",
    "teamName": "Mountainview Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/12.jpg",
    "title": "Mountainview Team",
  },
  {
    "favorite": true,
    "itemId": "50d39881-48af-488a-ac97-2c137105a392",
    "teamLogo": "https://robohash.org/animiquosquas.png?size=50x50&set=set1",
    "teamName": "Spaceship Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/14.jpg",
    "title": "Spaceship Team",
  },
  {
    "favorite": true,
    "itemId": "50068e33-2f0a-4679-9a52-29bfee839818",
    "teamLogo": "https://robohash.org/maximemaioreslaborum.png?size=50x50&set=set1",
    "teamName": "Koala Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/14.jpg",
    "title": "Koala Team",
  },
  {
    "favorite": false,
    "itemId": "5ce1c7ce-b6eb-4ce4-b182-ee493d4bf8ab",
    "teamLogo": "https://robohash.org/exercitationemdoloresdistinctio.png?size=50x50&set=set1",
    "teamName": "Pineapple Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/16.jpg",
    "title": "Pineapple Team",
  },
  {
    "favorite": true,
    "itemId": "6a408906-1db0-41ec-b445-472dd5be2035",
    "teamLogo": "https://robohash.org/etreiciendisperspiciatis.png?size=50x50&set=set1",
    "teamName": "Rock Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/18.jpg",
    "title": "Rock Team",
  },
  {
    "favorite": false,
    "itemId": "aee2a69c-67e3-4e0a-81d1-9b3435185786",
    "teamLogo": "https://robohash.org/enimquiet.png?size=50x50&set=set1",
    "teamName": "Summer Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/20.jpg",
    "title": "Summer Team",
  },
  {
    "favorite": false,
    "itemId": "633354ab-e6eb-40fa-a464-a04bc2e36b3a",
    "teamLogo": "https://robohash.org/corruptiminimaex.png?size=50x50&set=set1",
    "teamName": "Moonwalk Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/22.jpg",
    "title": "Moonwalk Team",
  },
  {
    "favorite": true,
    "itemId": "f2d507fe-682c-476a-9a91-56923d9e5a49",
    "teamLogo": "https://robohash.org/laboriosamdoloremdignissimos.png?size=50x50&set=set1",
    "teamName": "Emerald Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/24.jpg",
    "title": "Emerald Team",
  },
  {
    "favorite": true,
    "itemId": "3572b444-7d87-4b41-8e96-19a48f33aa82",
    "teamLogo": "https://robohash.org/exreprehenderitet.png?size=50x50&set=set1",
    "teamName": "Golden Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/26.jpg",
    "title": "Golden Team",
  },
  {
    "favorite": true,
    "itemId": "2c6f886f-de55-4ee8-b32d-57434df1076a",
    "teamLogo": "https://robohash.org/voluptatemetnesciunt.png?size=50x50&set=set1",
    "teamName": "Oceanfront Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/28.jpg",
    "title": "Oceanfront Team",
  },
  {
    "favorite": false,
    "itemId": "f52aebc4-3813-49c0-982d-ecf5192e2210",
    "teamLogo": "https://robohash.org/quiatotamesse.png?size=50x50&set=set1",
    "teamName": "Oakland Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/30.jpg",
    "title": "Oakland Team",
  },
  {
    "favorite": false,
    "itemId": "64bfdbfc-2e78-4dfc-b291-5666495e5cf6",
    "teamLogo": "https://robohash.org/remautemdolor.png?size=50x50&set=set1",
    "teamName": "Kapibara Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/32.jpg",
    "title": "Kapibara Team",
  },
  {
    "favorite": true,
    "itemId": "164129cc-7d04-4aa3-807f-ace25f9cf56c",
    "teamLogo": "https://robohash.org/evenieterrorquaerat.png?size=50x50&set=set1",
    "teamName": "Caterpillar Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/34.jpg",
    "title": "Caterpillar Team",
  },
  {
    "favorite": true,
    "itemId": "f5c4bed7-deb6-4144-b1d7-8c4b8732a192",
    "teamLogo": "https://robohash.org/reprehenderitrerumtempore.png?size=50x50&set=set1",
    "teamName": "Rain Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/36.jpg",
    "title": "Rain Team",
  },
  {
    "favorite": true,
    "itemId": "77b996e5-3c89-46bb-b1b2-40952e1362e7",
    "teamLogo": "https://robohash.org/dolordoloremlibero.png?size=50x50&set=set1",
    "teamName": "Zebra Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/38.jpg",
    "title": "Zebra Team",
  },
  {
    "favorite": false,
    "itemId": "d1b38724-b542-4b40-91ab-642b1d355365",
    "teamLogo": "https://robohash.org/noniuredoloribus.png?size=50x50&set=set1",
    "teamName": "Fireworks Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/40.jpg",
    "title": "Fireworks Team",
  },
  {
    "favorite": false,
    "itemId": "1c12c40c-6f1a-4805-8d46-d2f4ad237502",
    "teamLogo": "https://robohash.org/recusandaedolorumratione.png?size=50x50&set=set1",
    "teamName": "Air Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/42.jpg",
    "title": "Air Team",
  },
  {
    "favorite": true,
    "itemId": "30a83041-daf5-4fcb-8a8b-c28a5f193d8f",
    "teamLogo": "https://robohash.org/voluptatemquiest.png?size=50x50&set=set1",
    "teamName": "Turquoise Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/44.jpg",
    "title": "Turquoise Team",
  },
  {
    "favorite": true,
    "itemId": "b65a218c-0f3f-499c-ae4c-a3e6cbed201c",
    "teamLogo": "https://robohash.org/temporamolestiasesse.png?size=50x50&set=set1",
    "teamName": "Mark Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/46.jpg",
    "title": "Mark Team",
  },
  {
    "favorite": true,
    "itemId": "8b77b926-a229-4322-8b38-38a9f210eda7",
    "teamLogo": "https://robohash.org/rerumaliquamaut.png?size=50x50&set=set1",
    "teamName": "Butterfly Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/48.jpg",
    "title": "Butterfly Team",
  },
  {
    "favorite": false,
    "itemId": "a23b5086-33ed-4ad4-9ff0-a8a5f7650fcb",
    "teamLogo": "https://robohash.org/facilisetomnis.png?size=50x50&set=set1",
    "teamName": "Lion Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/50.jpg",
    "title": "Lion Team",
  },
  {
    "favorite": true,
    "itemId": "6439a585-54be-4362-ac97-ca43d1654026",
    "teamLogo": "https://robohash.org/etvoluptasquia.png?size=50x50&set=set1",
    "teamName": "Cat Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/52.jpg",
    "title": "Cat Team",
  },
  {
    "favorite": false,
    "itemId": "ba06d75f-ae4e-4d1d-9ffe-7664a6f1412d",
    "teamLogo": "https://robohash.org/quoculpaid.png?size=50x50&set=set1",
    "teamName": "Comet Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/54.jpg",
    "title": "Comet Team",
  },
  {
    "favorite": true,
    "itemId": "00a31331-b356-4f94-aab5-b0e8a4312740",
    "teamLogo": "https://robohash.org/sitaspernatureius.png?size=50x50&set=set1",
    "teamName": "Sailor Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/1.jpg",
    "title": "Sailor Team",
  },
  {
    "favorite": false,
    "itemId": "475a0ae4-b7bc-4a4e-83af-f370c9c6ad45",
    "teamLogo": "https://robohash.org/providentrerumeaque.png?size=50x50&set=set1",
    "teamName": "Panda Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/3.jpg",
    "title": "Panda Team",
  },
  {
    "favorite": true,
    "itemId": "b49128cc-fdbb-410d-95b0-ffa48ae0c314",
    "teamLogo": "https://robohash.org/estfacereet.png?size=50x50&set=set1",
    "teamName": "Riverside Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/5.jpg",
    "title": "Riverside Team",
  },
  {
    "favorite": false,
    "itemId": "0b61fb70-65e6-4730-9cf8-36016a8de80f",
    "teamLogo": "https://robohash.org/occaecatibeataemaxime.png?size=50x50&set=set1",
    "teamName": "Sunflower Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/7.jpg",
    "title": "Sunflower Team",
  },
  {
    "favorite": undefined,
    "itemId": "empty-optional-fields",
    "teamLogo": "https://robohash.org/iureinipsam.png?size=50x50&set=set1",
    "teamName": "Commando Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/11.jpg",
    "title": "Commando Team",
  },
]
`;

exports[`fetchTeams useTeams should return all teams data and no error on successful fetch 1`] = `
[
  {
    "favorite": true,
    "itemId": "be00b94a-e12a-48c4-9615-23c3478f1bd0",
    "teamLogo": "https://robohash.org/iureinipsam.png?size=50x50&set=set1",
    "teamName": "Silver Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/6.jpg",
    "title": "Silver Team",
  },
  {
    "favorite": true,
    "itemId": "fdd82ada-af3f-4fe8-a90b-254877082e97",
    "teamLogo": "https://robohash.org/odiotemporeaccusamus.png?size=50x50&set=set1",
    "teamName": "Sunset Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/8.jpg",
    "title": "Sunset Team",
  },
  {
    "favorite": false,
    "itemId": "635e3d02-f128-460d-a630-628fec86ef2e",
    "teamLogo": "https://robohash.org/omnisadsequi.png?size=50x50&set=set1",
    "teamName": "Liberty Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/10.jpg",
    "title": "Liberty Team",
  },
  {
    "favorite": true,
    "itemId": "f6dd98e2-d7d2-48bb-ab4d-251a0c21255a",
    "teamLogo": "https://robohash.org/molestiaevoluptasdeserunt.png?size=50x50&set=set1",
    "teamName": "Mountainview Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/12.jpg",
    "title": "Mountainview Team",
  },
  {
    "favorite": true,
    "itemId": "50d39881-48af-488a-ac97-2c137105a392",
    "teamLogo": "https://robohash.org/animiquosquas.png?size=50x50&set=set1",
    "teamName": "Spaceship Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/14.jpg",
    "title": "Spaceship Team",
  },
  {
    "favorite": true,
    "itemId": "50068e33-2f0a-4679-9a52-29bfee839818",
    "teamLogo": "https://robohash.org/maximemaioreslaborum.png?size=50x50&set=set1",
    "teamName": "Koala Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/14.jpg",
    "title": "Koala Team",
  },
  {
    "favorite": false,
    "itemId": "5ce1c7ce-b6eb-4ce4-b182-ee493d4bf8ab",
    "teamLogo": "https://robohash.org/exercitationemdoloresdistinctio.png?size=50x50&set=set1",
    "teamName": "Pineapple Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/16.jpg",
    "title": "Pineapple Team",
  },
  {
    "favorite": true,
    "itemId": "6a408906-1db0-41ec-b445-472dd5be2035",
    "teamLogo": "https://robohash.org/etreiciendisperspiciatis.png?size=50x50&set=set1",
    "teamName": "Rock Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/18.jpg",
    "title": "Rock Team",
  },
  {
    "favorite": false,
    "itemId": "aee2a69c-67e3-4e0a-81d1-9b3435185786",
    "teamLogo": "https://robohash.org/enimquiet.png?size=50x50&set=set1",
    "teamName": "Summer Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/20.jpg",
    "title": "Summer Team",
  },
  {
    "favorite": false,
    "itemId": "633354ab-e6eb-40fa-a464-a04bc2e36b3a",
    "teamLogo": "https://robohash.org/corruptiminimaex.png?size=50x50&set=set1",
    "teamName": "Moonwalk Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/22.jpg",
    "title": "Moonwalk Team",
  },
  {
    "favorite": true,
    "itemId": "f2d507fe-682c-476a-9a91-56923d9e5a49",
    "teamLogo": "https://robohash.org/laboriosamdoloremdignissimos.png?size=50x50&set=set1",
    "teamName": "Emerald Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/24.jpg",
    "title": "Emerald Team",
  },
  {
    "favorite": true,
    "itemId": "3572b444-7d87-4b41-8e96-19a48f33aa82",
    "teamLogo": "https://robohash.org/exreprehenderitet.png?size=50x50&set=set1",
    "teamName": "Golden Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/26.jpg",
    "title": "Golden Team",
  },
  {
    "favorite": true,
    "itemId": "2c6f886f-de55-4ee8-b32d-57434df1076a",
    "teamLogo": "https://robohash.org/voluptatemetnesciunt.png?size=50x50&set=set1",
    "teamName": "Oceanfront Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/28.jpg",
    "title": "Oceanfront Team",
  },
  {
    "favorite": false,
    "itemId": "f52aebc4-3813-49c0-982d-ecf5192e2210",
    "teamLogo": "https://robohash.org/quiatotamesse.png?size=50x50&set=set1",
    "teamName": "Oakland Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/30.jpg",
    "title": "Oakland Team",
  },
  {
    "favorite": false,
    "itemId": "64bfdbfc-2e78-4dfc-b291-5666495e5cf6",
    "teamLogo": "https://robohash.org/remautemdolor.png?size=50x50&set=set1",
    "teamName": "Kapibara Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/32.jpg",
    "title": "Kapibara Team",
  },
  {
    "favorite": true,
    "itemId": "164129cc-7d04-4aa3-807f-ace25f9cf56c",
    "teamLogo": "https://robohash.org/evenieterrorquaerat.png?size=50x50&set=set1",
    "teamName": "Caterpillar Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/34.jpg",
    "title": "Caterpillar Team",
  },
  {
    "favorite": true,
    "itemId": "f5c4bed7-deb6-4144-b1d7-8c4b8732a192",
    "teamLogo": "https://robohash.org/reprehenderitrerumtempore.png?size=50x50&set=set1",
    "teamName": "Rain Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/36.jpg",
    "title": "Rain Team",
  },
  {
    "favorite": true,
    "itemId": "77b996e5-3c89-46bb-b1b2-40952e1362e7",
    "teamLogo": "https://robohash.org/dolordoloremlibero.png?size=50x50&set=set1",
    "teamName": "Zebra Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/38.jpg",
    "title": "Zebra Team",
  },
  {
    "favorite": false,
    "itemId": "d1b38724-b542-4b40-91ab-642b1d355365",
    "teamLogo": "https://robohash.org/noniuredoloribus.png?size=50x50&set=set1",
    "teamName": "Fireworks Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/40.jpg",
    "title": "Fireworks Team",
  },
  {
    "favorite": false,
    "itemId": "1c12c40c-6f1a-4805-8d46-d2f4ad237502",
    "teamLogo": "https://robohash.org/recusandaedolorumratione.png?size=50x50&set=set1",
    "teamName": "Air Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/42.jpg",
    "title": "Air Team",
  },
  {
    "favorite": true,
    "itemId": "30a83041-daf5-4fcb-8a8b-c28a5f193d8f",
    "teamLogo": "https://robohash.org/voluptatemquiest.png?size=50x50&set=set1",
    "teamName": "Turquoise Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/44.jpg",
    "title": "Turquoise Team",
  },
  {
    "favorite": true,
    "itemId": "b65a218c-0f3f-499c-ae4c-a3e6cbed201c",
    "teamLogo": "https://robohash.org/temporamolestiasesse.png?size=50x50&set=set1",
    "teamName": "Mark Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/46.jpg",
    "title": "Mark Team",
  },
  {
    "favorite": true,
    "itemId": "8b77b926-a229-4322-8b38-38a9f210eda7",
    "teamLogo": "https://robohash.org/rerumaliquamaut.png?size=50x50&set=set1",
    "teamName": "Butterfly Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/48.jpg",
    "title": "Butterfly Team",
  },
  {
    "favorite": false,
    "itemId": "a23b5086-33ed-4ad4-9ff0-a8a5f7650fcb",
    "teamLogo": "https://robohash.org/facilisetomnis.png?size=50x50&set=set1",
    "teamName": "Lion Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/50.jpg",
    "title": "Lion Team",
  },
  {
    "favorite": true,
    "itemId": "6439a585-54be-4362-ac97-ca43d1654026",
    "teamLogo": "https://robohash.org/etvoluptasquia.png?size=50x50&set=set1",
    "teamName": "Cat Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/52.jpg",
    "title": "Cat Team",
  },
  {
    "favorite": false,
    "itemId": "ba06d75f-ae4e-4d1d-9ffe-7664a6f1412d",
    "teamLogo": "https://robohash.org/quoculpaid.png?size=50x50&set=set1",
    "teamName": "Comet Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/54.jpg",
    "title": "Comet Team",
  },
  {
    "favorite": true,
    "itemId": "00a31331-b356-4f94-aab5-b0e8a4312740",
    "teamLogo": "https://robohash.org/sitaspernatureius.png?size=50x50&set=set1",
    "teamName": "Sailor Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/1.jpg",
    "title": "Sailor Team",
  },
  {
    "favorite": false,
    "itemId": "475a0ae4-b7bc-4a4e-83af-f370c9c6ad45",
    "teamLogo": "https://robohash.org/providentrerumeaque.png?size=50x50&set=set1",
    "teamName": "Panda Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/3.jpg",
    "title": "Panda Team",
  },
  {
    "favorite": true,
    "itemId": "b49128cc-fdbb-410d-95b0-ffa48ae0c314",
    "teamLogo": "https://robohash.org/estfacereet.png?size=50x50&set=set1",
    "teamName": "Riverside Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/5.jpg",
    "title": "Riverside Team",
  },
  {
    "favorite": false,
    "itemId": "0b61fb70-65e6-4730-9cf8-36016a8de80f",
    "teamLogo": "https://robohash.org/occaecatibeataemaxime.png?size=50x50&set=set1",
    "teamName": "Sunflower Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/7.jpg",
    "title": "Sunflower Team",
  },
  {
    "favorite": undefined,
    "itemId": "empty-optional-fields",
    "teamLogo": "https://robohash.org/iureinipsam.png?size=50x50&set=set1",
    "teamName": "Commando Team",
    "thumbnail": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/11.jpg",
    "title": "Commando Team",
  },
]
`;
